std/experimental/color/rgb.d

```diff
@@ -0,0 +1,852 @@
+module std.experimental.color.rgb;
+
+import std.experimental.color;
+import std.experimental.color.xyz;
+
+import std.traits: isInstanceOf, isNumeric, isIntegral, isFloatingPoint, isSigned, isSomeChar, Unqual;
+import std.typetuple: TypeTuple;
+import std.typecons: tuple;
+
+@safe: pure: nothrow: @nogc:
+
+
+enum isRGB(T) = isInstanceOf!(RGB, T);
+
+enum RGBColorSpace
+{
+    sRGB,
+    sRGB_Gamma2_2,
+/+ TODO: these all need testing
+    AdobeRGB,
+    AppleRGB,
+    BestRGB,
+    BetaRGB,
+    BruceRGB,
+    CIE_RGB,
+    ColorMatchRGB,
+    DonRGB4,
+//    ECI_RGB_v2,
+    EktaSpacePS5,
+    NTSC_RGB,
+    PAL_SECAM_RGB,
+    ProPhotoRGB,
+    SMPTE_C_RGB,
+    WideGamutRGB
++/
+}
+
+
+struct RGB(string components_, ComponentType_, bool linear_ = false, RGBColorSpace colorSpace_ = RGBColorSpace.sRGB) if(isValidComponentType!ComponentType_)
```

Comment:
Well, not really. Colour spaces strictly specify their particular gamma ramp.
An RGB colourspace definition is a combination of RGB axiis, a whitepoint and a gamma scheme. These facts are all immutable wrt a particular colour space (the table is down lower).
The only useful option on this is whether the gamma is applied or not, because it means the difference between an encoded colour (by definition), or the ability to sum colours or perform colour space conversions (which are linear operations), and both states are useful to all colour spaces.